### ISSUE TITLE
Implementation of interface requires members redeclaration - inside definition file

### ISSUE THREAD

**Issue by CzBuCHi** [Microsoft/TypeScript#16303]
**TypeScript Version:**  2.3.2
**Code:**

```ts
// external_library.d.ts
interface A {
    foo: Function;
}
declare class B implements A { 
}
```

**Expected behavior:** no error
**Actual behavior:** error: Class 'B' incorrectly implements interface 'A'

This is sort of duplicate of #5749 or #340 except class is declared in **d.ts** file and implemented by javascript

**Why:** Currently is needed to copy all declarations from A into B... -> duplicate code

**Comment by RyanCavanaugh** [comment]
Not sure what makes this a not-exact-duplicate of #340. See https://github.com/Microsoft/TypeScript/issues/340#issuecomment-184964440

**Comment by CzBuCHi** [comment]
@RyanCavanaugh sorry about that - i missed that comment ...
